### PR TITLE
fix(a11y): improve challenge test suite results accessibility

### PR DESCRIPTION
**Pull request by bbsmooth**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes locally on my machine.

Closes #XXXXX

The current test results displayed under the Get Help button are using the `aria-label` attribute on a `<div>` to present the results to screen readers. Unfortunately, the use of this attribute on a generic element like a `<div>` is not well supported and sure enough the JAWS screen reader completely ignores these divs. I have redone the HTML so that screen reader users will read the actual content on the page. They will get an sr-only status message before each test of either "Waiting", "Passed", or "Failed". Also, I have moved the content into a list so that the screen reader will announce how many tests are in the list. Finally, I added the h2 heading "Tests" above the list, which I think will make it easier for screen reader users to navigate to these. If we don't want the header to be shown on the page then we can visually hide it.

### CODE REVIEW

File: client/src/templates/Challenges/components/test-suite.tsx [hunk 1]
```diff
@@ -20,47 +20,37 @@ interface TestSuiteProps {
...
-      })}
-    </div>
+    <>
+      <h2 className='challenge-test-suite-heading'>Tests</h2>
+      <ul className='challenge-test-suite'>
+        {testSuiteTests.map(({ err, pass = false, text = '' }, index) => {
+          const isInitial = !pass && !err;
+          const statusIcon = pass && !err ? <GreenPass /> : <Fail />;
+          const initialText = 'Waiting';
+          const statusText = pass && !err ? 'Passed' : 'Failed';
```
Comment (moT01) [inline client/src/templates/Challenges/components/test-suite.tsx:34]: Maybe outside the scope of this PR, but shouldn't we put these in the translations?

**Review by moT01 — COMMENTED** [review]
I wasn't sure on the 'tests' header at first, but I think I like it.

One thing I noticed wasn't that I couldn't reach that tests area with the tab key, it jumps over to the editor or other places when hitting tab while the help button is highlighted. I didn't have any issues using the voice over hotkeys. Curious @bbsmooth - do you know if screen reader users use tab often to navigate?

I left one other concern below - I don't know if any of this needs to be adjusted here. I would be fine merging it like this since it's a good improvement and my concerns were all pre-existing and might be outside the scope of what is trying to be accomplished here.

**Comment by moT01** [comment]
That test failed again? It passed twice for me locally. hmm.

**Comment by moT01** [comment]
Well, I pushed a commit to try and fix cypress - I couldn't reproduce the fail locally, so I'm not sure if it will work. Fingers crossed.

File: client/src/templates/Challenges/components/test-suite.tsx (hunk 1 above)
Comment (moT01) [reply client/src/templates/Challenges/components/test-suite.tsx]: added

**Comment by moT01** [comment]
I hope you don't mind me working on this @bbsmooth - but I went ahead and added that text to the translations as well, since I had this up and running.

**Comment by bbsmooth** [comment]
> I hope you don't mind me working on this

Not at all. Those translations were just something I overlooked and definitely needed to be done.

> One thing I noticed wasn't that I couldn't reach that tests area with the tab key ... do you know if screen reader users use tab often to navigate?

The Tab key is used to navigate between functional elements, such as inputs and buttons. It is not used for reading regular text. Screen reader users have a variety of ways to browse the text, we do not, and should not add tab stops to "help" them.

One thing that does need to be done is to add live feedback for screen reader users so they know that the tests have actually been run and completed. I am going to do that in a separate PR after we get this one finished. I didn't want to make this one too complicated with the hopes it would move faster.

**Review by moT01 — APPROVED** [review]
got it ✅ LGTM 🎉

**Review by raisedadead — APPROVED** [review]

**Pull request merged**